### PR TITLE
PiTR: support empty schema in PiTR.

### PR DESCRIPTION
**Pull request by joccau**
Signed-off-by: joccau <zak.zhao@pingcap.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/33866

Problem Summary:

### What is changed and how it works?
Support empty schema in PiTR.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

### test
- backup steps
    - create empty schema `emptydb`
    - `br backup full` in up-stream
    - `br log start ` in up-stream
    - create table `emptydb.t` and insert 1000 record.
   
- restore steps 1
- `br restore point --full-backup-storage --s ` in down-stream.   `emptydb.t` and records has been recovered.

- restore steps 2
- `br restore full` in down-stream
- `br restore point --start-ts --restore-ts ` in down-stream. `emptydb.t` and records has been recovered.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/utils/json.go [hunk 1]
```diff
@@ -103,8 +103,11 @@ func makeJSONSchema(schema *backuppb.Schema) (*jsonSchema, error) {
 	if err := json.Unmarshal(schema.Db, &result.DB); err != nil {
 		return nil, errors.Trace(err)
 	}
-	if err := json.Unmarshal(schema.Table, &result.Table); err != nil {
-		return nil, errors.Trace(err)
+
+	if schema.Table != nil {
```
Comment (3pointer) [inline br/pkg/utils/json.go:107]: shall we add the same check in L119?

Comment (joccau) [reply br/pkg/utils/json.go:107]: Have Modified it.

**Review by 3pointer — APPROVED** [review]

**Pull request merged**